### ISSUE TITLE
Optionally relativize paths in CodeClimate output

### ISSUE THREAD

**Issue by fgabolde** [tomasbjerre/violations-lib#123]
tl;dr: I'd like to have a way to relativize paths either in violations-lib or in violations-maven-plugin, possibly only for the CodeClimate output, because checkstyle won't let me

I'm using the Maven plugin to convert checkstyle violations so I can have code quality reports in GitLab, which expects CodeClimate JSON. It works fine except that we output absolute paths in the Checkstyle XML, so the report has absolute paths too, which can't be matched to files in the repository because they point to files on the runner e.g. `/builds/some/repository/src/main/java/foo/bar/baz/SomeClass.java:86`.

The problem with Checkstyle is that it is a pain to configure for relative paths. Maven's Checkstyle plugin does have a `checkstyle.basedir` option, but that only works for generating HTML reports which is too late for the XML report used by violations-lib. You can add a `basedir` property directly in the checkstyle configuration like this:

```xml
<module name = "Checker">
  <property name="basedir" value="${checkstyle.basedir}"/>
  <module name="LineLength">
    ...
```

and that will work, but from now on ALL users of the configuration MUST specify a basedir property and they are stuck with relative paths (there is no possible value for `basedir` that will go back to absolute paths) even if they don't want to. In addition, this requires modifying the checkstyle config, which is awkward for users of the plain built-in google_checks.xml or sun_checks.xml.

**Comment by tomasbjerre** [comment]
releasing a fix for this, open issue again if not working.

**Comment by fgabolde** [comment]
@tomasbjerre Thanks a lot, I was able to upgrade to violations-maven-plugin 1.50.4 which I assume includes this, and paths now start at the repository root which is exactly what I needed.
